### PR TITLE
Feature/11/setting up routes for kid

### PR DESCRIPTION
**Pull request by Skyline857**
Feature/backend: set up routes for Kid[#11]

Requirements
Add routes for managing Kid, use the requirements in GitBook  [(https://victorias-organization-2.gitbook.io/kids-first-app/product-details/kids-info/adding-editing-deleting  l)

Test Steps
Set Up Postman:
Open Postman to test the API endpoints.

Testing Instructions for Postman

1. Create a New Kid Profile (POST /kid)
Endpoint: POST http://localhost:3000/kid
Headers:
Content-Type: application/json
Body (JSON):
json
{
  "name": "John",
  "childColor": "blue",
  "DOB": "01/01/2015",
  "allergies": ["Peanuts"],
  "interest": ["Soccer"],
  "fears": ["Dogs"],
  "otherNotes": "Loves to play soccer"
}
Expected Response:
Status: 201 Created
Body: The details of the created kid profile, e.g.:
json
{
  "_id": "60d2b8f2e8f8c2d1e8f8c2d1",
  "name": "John",
  "childColor": "blue",
  "DOB": "01/01/2015",
  "allergies": ["Peanuts"],
  "interest": ["Soccer"],
  "fears": ["Dogs"],
  "otherNotes": "Loves to play soccer",
  "createdAt": "2024-07-01T00:00:00.000Z",
  "updatedAt": "2024-07-01T00:00:00.000Z"
}

2. Update an Existing Kid Profile (PUT /kid/
)
Endpoint: PUT http://localhost:3000/kid/:id
Replace :id with the actual ID of the kid profile you want to update.
Headers:
Content-Type: application/json
Body (JSON):
json
{
  "name": "Johnny",
  "childColor": "green",
  "DOB": "01/01/2016",
  "allergies": ["Dust"],
  "interest": ["Reading"],
  "fears": ["Heights"],
  "otherNotes": "Loves reading books"
}
Expected Response:
Status: 200 OK
Body: The updated kid profile details, e.g.:
json
{
  "_id": "60d2b8f2e8f8c2d1e8f8c2d1",
  "name": "Johnny",
  "childColor": "green",
  "DOB": "01/01/2016",
  "allergies": ["Dust"],
  "interest": ["Reading"],
  "fears": ["Heights"],
  "otherNotes": "Loves reading books",
  "createdAt": "2024-07-01T00:00:00.000Z",
  "updatedAt": "2024-07-02T00:00:00.000Z"
}

3. Get All Kid Profiles (GET /kids)
Endpoint: GET http://localhost:3000/kids
Expected Response:
Status: 200 OK
Body: An array of all kid profiles, e.g.:
json
[
  {
    "_id": "60d2b8f2e8f8c2d1e8f8c2d1",
    "name": "John",
    "childColor": "blue",
    "DOB": "01/01/2015",
    "allergies": ["Peanuts"],
    "interest": ["Soccer"],
    "fears": ["Dogs"],
    "otherNotes": "Loves to play soccer",
    "createdAt": "2024-07-01T00:00:00.000Z",
    "updatedAt": "2024-07-01T00:00:00.000Z"
  },
  {
    "_id": "60d2b8f2e8f8c2d1e8f8c2d2",
    "name": "Jane",
    "childColor": "red",
    "DOB": "01/01/2014",
    "allergies": ["None"],
    "interest": ["Painting"],
    "fears": ["Spiders"],
    "otherNotes": "Loves painting",
    "createdAt": "2024-07-01T00:00:00.000Z",
    "updatedAt": "2024-07-01T00:00:00.000Z"
  }
]

4. Get a Specific Kid Profile by ID (GET /kid/
)
Endpoint: GET http://localhost:3000/kid/:id
Replace :id with the actual ID of the kid profile you want to retrieve.
Expected Response:
Status: 200 OK
Body: The details of the specific kid profile, e.g.:
json
{
  "_id": "60d2b8f2e8f8c2d1e8f8c2d1",
  "name": "John",
  "childColor": "blue",
  "DOB": "01/01/2015",
  "allergies": ["Peanuts"],
  "interest": ["Soccer"],
  "fears": ["Dogs"],
  "otherNotes": "Loves to play soccer",
  "createdAt": "2024-07-01T00:00:00.000Z",
  "updatedAt": "2024-07-01T00:00:00.000Z"
}

5. Delete a Kid Profile by ID (DELETE /kid/
)
Endpoint: DELETE http://localhost:3000/kid/:id
Replace :id with the actual ID of the kid profile you want to delete.
Expected Response:
Status: 204 No Content

API Changes
New Endpoints Added:
POST /kid: Create a new kid profile.
PUT /kid/:id: Update an existing kid profile.
GET /kids: Get all kid profiles.
GET /kid/:id: Get a specific kid profile.
DELETE /kid/:id: Delete a specific kid profile.

Database Changes
Schema: Ensure the kid schema includes fields like name, childColor, DOB, allergies, interest, fears, and otherNotes.

Checklist
 Pull Request title includes the issue number and a brief description of the change.
 All changes should be tested and verified to work correctly.
 Code follows backend best practices.
 Branch names follow the conventions in the contributing file.
 Commit messages follow the naming conventions in the contributing file.
 No sensitive data or secrets are included in the codebase.

### CODE REVIEW

File: routes/kids.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+const router = express.Router();
+const {
+    createKid,
+    updateKid,
+    getAllKids,
+    deleteKid,
+} = require('../controllers/kid-controller');  
+
+
+router.post('/kid', kidController.kid);
```
Comment (ElenaCherpakova) [inline routes/kids.js]: It should be /kids/:id to follow RESTful conventions for resources. Using the plural form (kids) is a common practice as it represents the collection of all kids, and :id refers to a specific item within that collection
For reference: https://restfulapi.net/resource-naming/

**Review by ElenaCherpakova — CHANGES_REQUESTED** [review]

File: routes/kids.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+const router = express.Router();
+const {
+    createKid,
```
Comment (AnnaSolovykh) [inline routes/kids.js]: Where is this function used?

File: routes/kids.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    getAllKids,
+    deleteKid,
+} = require('../controllers/kid-controller');  
+
+
+router.post('/kid', kidController.kid);
+router.put('/kid/:id', kidController.updateKid);
+router.delete('/kid/:id', kidController.deleteKid);            
+router.get('/kid', kidController.getAllKids);
+router.get('/kid/:id', kidController.getKid);
```
Comment (AnnaSolovykh) [inline routes/kids.js]: Where is this function imported?

File: routes/kids.js (hunk 1 above)
Comment (Skyline857) [reply routes/kids.js]: Thank you for such a detailed response, Lena! I appreciate it a lot!

File: routes/kids.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const {
 } = require('../controllers/kid-controller');  
 
 
-router.post('/kid', kidController.kid);
+router.post('/kids:id', kidController.createKid);
```
Comment (ElenaCherpakova) [inline routes/kids.js]: there are still issues with routes it should be` router.post('/kids/:id', kidController.updateKid);`

File: routes/kids.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ const {
...
 router.put('/kid/:id', kidController.updateKid);
```
Comment (ElenaCherpakova) [inline routes/kids.js]: same here

**Review by AnnaSolovykh — APPROVED** [review]

**Review by ElenaCherpakova — APPROVED** [review]

**Pull request merged**